spmd/tensor/parallel/api.py

```diff
@@ -76,12 +76,69 @@ def _shard_self_attn_params(name: str, module: nn.Module) -> None:
                 module.register_module(n, tp_multi_head_attention)
 
 
+def _isMLP(module: nn.Module) -> bool:
+    # We assume that the structure of MLP to shard defined in object way as below:
+    # Linear -> ReLU -> Linear
+    submodules = list(module.children())
+    return (
```

Comment:
Isn't this too strong? We really don't care what's in the middle. At end of the day, users can have many submodules, we just want to verify that there are two Linear or even numbers of Linear. This might have drawback, for example, what if user call layer norm after the first layer, but that's fine for now and DTensor does cover these ops already.